lib/jobs/snmp-job.js

```diff
@@ -151,15 +152,20 @@ function snmpJobFactory(
             self._done(err);
         });
     };
-
+    
     /**
      * @memberOf SnmpJob
      */
     SnmpJob.prototype._collectMetricData = function(data) {
+        var self = this;
         assert.object(data);
         assert.object(data.config);
         assert.string(data.config.metric);
         var Metric, metric;
+        var dataCopy = _.cloneDeep(data);
+        dataCopy.cache = self.resultCache;
+        dataCopy.routingKey = self.routingKey;
```

Comment:
`dataCopy` is passed into `collectMetricData()`.  Both `cache` and `routingKey` values are used there.